StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerSK2Tests.swift

```diff
@@ -131,4 +133,122 @@ class TrialOrIntroPriceEligibilityCheckerSK2Tests: StoreKitConfigTestCase {
             expect(receivedEligibility.status) == expected[product]
         }
     }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
```

Comment:
for other readers: Cody and I synced up about this: we don't have a way to also test this for Sk1 directly. 
The issue is that we don't have a mechanism (currently) for us to choose the SK1 implementation over the SK2 one. 
So, for devices that support SK2, it will be used automatically. 

For the other method, what we did was to call the sk1 implementation directly... but that would bypass the new method entirely, which defeats the point. 

This should be refactored, I think ideally to the point where we have one class that makes the decision of which implementation to use (backend, sk1 or sk2) and one class per implementation (something like the GOF Strategy pattern comes to mind), and then we'd be able to have clear tests for each implementation separately. 

But that's way out of scope for this PR, so for now, we're just testing the method through SK2.

Reply:
Makes sense. The compiler `if #available` directives makes that approach harder because we'd have to still add those at the top level.